codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ApplicationProtocol.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen;
+
+import java.util.Objects;
+import software.amazon.smithy.codegen.core.Symbol;
+
+
+/**
+ * Represents the resolves {@link Symbol}s and references for an
+ * application protocol (e.g., "http", "mqtt", etc).
+ */
+public final class ApplicationProtocol {
+
+    private final String name;
+    private final Symbol requestType;
+    private final Symbol responseType;
+
+    /**
+     * Creates a resolved application protocol.
+     *  @param name         The protocol name (e.g., http, mqtt, etc).
```

Comment:
Nit: extra space between * and param@

Reply:
This lines up the descriptions of the paramters

Reply:
Sorry should have pointed out that they are two spaces instead of one between * and param@. But it's a nitpick, and doesn't really matter